Doppler.AccountPlans/Controllers/AccountPlansController.cs

```diff
@@ -46,11 +51,18 @@ public async Task<IActionResult> GetCalculateUpgradeCost([FromRoute] string acco
             if (newPlan == null)
                 return new NotFoundResult();
 
-            var upgradeCost = CalculateUpgradeCostHelper.CalculatePlanAmountDetails(newPlan, discountPlan, currentPlan, _dateTimeProvider.Now);
+            var promotion = new Promotion();
+            if (!string.IsNullOrEmpty(promocode))
+            {
+                var encryptedCode = _encryptionService.EncryptAES256(promocode);
+                promotion = await _promotionRepository.GetPromotionByCode(encryptedCode, newPlanId);
+            }
+
+            var upgradeCost = CalculateUpgradeCostHelper.CalculatePlanAmountDetails(newPlan, discountPlan, currentPlan, _dateTimeProvider.Now, promotion);
+
             return new OkObjectResult(upgradeCost);
         }
 
-        [Authorize(Policies.OWN_RESOURCE_OR_SUPERUSER)]
```

Comment:
Is it ok to remove this authorize validation? Is not the promocode related to an user?

Reply:
It is ok because this resource does not have an account name in the URL 
` [HttpGet("/plans/{planId}/validate/{promocode}")]`
Is not necessary to check if the user is a Superuser

Reply:
This endpoint only return information of the promo code by plan, discount, credits, etc., But nothing of the user